test/unit/Config.test.ts

```diff
@@ -1,16 +1,43 @@
 import { arrayify, BytesLike } from '@ethersproject/bytes'
 import { StreamrClient } from '../../src/StreamrClient'
+import set from 'lodash.set'
 
-const createClient = (privateKey: BytesLike) => {
-    return new StreamrClient({
-        auth: {
-            privateKey
+describe('Config', () => {
+
+    describe('validate ethereum addresses', () => {
+        const createClient = (propertyPaths: string, value: string|undefined|null) => {
+            const opts: any = {}
+            set(opts, propertyPaths, value)
+            return new StreamrClient(opts)
+        }
+        const propertyPaths: string[] = [
+            'streamrNodeAddress',
+            'tokenAddress',
+            'tokenSidechainAddress',
+            'dataUnion.factoryMainnetAddress',
+            'dataUnion.factorySidechainAddress',
+            'dataUnion.templateMainnetAddress',
+            'dataUnion.templateSidechainAddress'
+        ]
+        for (const propertyPath of propertyPaths) {
+            it(propertyPath, () => {
+                const errorMessage = `${propertyPath} is not a valid Ethereum address`
+                expect(() => createClient(propertyPath, 'invalid-address')).toThrow(errorMessage)
+                expect(() => createClient(propertyPath, undefined)).toThrow(errorMessage)
+                expect(() => createClient(propertyPath, null)).toThrow(errorMessage)
+                expect(() => createClient(propertyPath, '0x1234567890123456789012345678901234567890')).not.toThrow()
+            })
         }
     })
-}
 
-describe('Config', () => {
     describe('private key', () => {
+        const createClient = (privateKey: BytesLike) => {
```

Comment:
I get why you did this but feel like having two functions with the same name but different signatures isn't great. Maybe rename one of them or make it so one function can handle both cases?

Reply:
Renamed one of the methods. The method "overloading" was quite clear from the context of each test. As all tests in that `describe` block create the client same way, short method names would be enough to understand what is happening.